src/main/java/org/embulk/output/elasticsearch/ElasticsearchHttpClient.java

```diff
@@ -0,0 +1,368 @@
+        // Index found: 200
+        try {
+            sendRequest(indexName, HttpMethod.GET, task, retryHelper);
+            return true;
+        }
+        catch (ResourceNotFoundException ex) {
+            return false;
+        }
+    }
+
+    public String generateNewIndexName(String indexName)
+    {
+        Timestamp time = Exec.getTransactionTime();
+        return indexName + new SimpleDateFormat("_yyyyMMdd-HHmmss").format(time.toEpochMilli());
+    }
+
+    public boolean isAliasExisting(String aliasName, PluginTask task, Jetty92RetryHelper retryHelper)
+    {
+        // curl -XGET localhost:9200/_aliases  // List all aliases
+        // No aliases: {}
+        // Aliases found: {"embulk_20161018-183738":{"aliases":{"embulk":{}}}}
+        JsonNode response = sendRequest("/_aliases", HttpMethod.GET, task, retryHelper);
+        if (response.size() == 0) {
+            return false;
+        }
+        for (JsonNode index : response) {
+            if (index.has("aliases") && index.get("aliases").has(aliasName)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // Should be called just once while Embulk transaction.
+    // Be sure to call after all exporting tasks completed
+    // This method will delete existing index
+    public void reassignAlias(String aliasName, String newIndexName, PluginTask task, Jetty92RetryHelper retryHelper)
+    {
+        if (!isAliasExisting(aliasName, task, retryHelper)) {
+            assignAlias(newIndexName, aliasName, task, retryHelper);
+        }
+        else {
+            List<String> oldIndices = getIndexByAlias(aliasName, task, retryHelper);
+            assignAlias(newIndexName, aliasName, task, retryHelper);
+            for (String index : oldIndices) {
+                deleteIndex(index, task, retryHelper);
+            }
+        }
+    }
+
+    public String getEsVersion(PluginTask task, Jetty92RetryHelper retryHelper)
+    {
+        // curl -XGET 'http://localhost:9200’
+        JsonNode response = sendRequest("", HttpMethod.GET, task, retryHelper);
+        return response.get("version").get("number").asText();
+    }
+
+    public void validateIndexOrAliasName(String index, String type)
+    {
+        for (int i = 0; i < index.length(); i++) {
+            if (inalidIndexCharaters.contains(index.charAt(i))) {
+                throw new ConfigException(String.format("%s '%s' must not contain the invalid characters " + inalidIndexCharaters.toString(), type, index));
+            }
+        }
+
+        if (!index.toLowerCase(Locale.ROOT).equals(index)) {
+            throw new ConfigException(String.format("%s '%s' must be lowercase", type, index));
+        }
+
+        if (index.startsWith("_") || index.startsWith("-") || index.startsWith("+")) {
+            throw new ConfigException(String.format("%s '%s' must not start with '_', '-', or '+'", type, index));
+        }
+
+        if (index.length() > maxIndexNameBytes) {
```

Comment:
This validation is good for me but, just in case, I wanted to mention that we can use `@Min` and `@Max` in PluginTask to do that.